### PR TITLE
CONTRIBUTING and RELEASE updates

### PR DESCRIPTION
**Pull request by rhyolight**
The process in `RELEASE.md` is not implemented. We will discuss this Wednesday at Numenta HQ.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+# Release (that doesn't quite work yet)
+
+- Update version available via `nupic.torch.__version__`
+- `git tag $VERSION`
+- `git push --tags upstream`
```
Comment (lscheinkman) [inline RELEASE.md]: Dont forget to create a github release

File: .circleci/config.yml [hunk 2]
```diff
@@ -108,12 +107,32 @@ jobs:
...
+      #     name: Deploy to PYPI
+      #     environment:
+      #       VERBOSE: 1
+      #     command: python -m twine upload --repository-url https://test.pypi.org/legacy/ dist/*
 
 workflows:
   version: 2
 
-  validate-build-test:
+  validate-build-test-deploy:
```
Comment (lscheinkman) [inline .circleci/config.yml]: If you change the job name then you need to update gitbub branch protection rules:  See https://github.com/numenta/nupic.torch/settings/branch_protection_rules/5520686
I recommend not changing the name if possible.

Comment (rhyolight) [reply .circleci/config.yml]: this may be my whole problem

File: .circleci/config.yml [hunk 3]
```diff
@@ -17,34 +17,28 @@
...
       - run:
           name: Validate contributor license
           command: ci/validate-contributor.sh
-      - save_cache:
-          key: source-cache-{{ .Environment.CIRCLE_SHA1 }}
-          paths:
-            - ~/nupic.torch
+      - persist_to_workspace:
+          root: ~/nta
+          paths: nupic.torch
```
Comment (rhyolight) [inline .circleci/config.yml]: this is so much better 👍

File: .circleci/config.yml [hunk 4]
```diff
@@ -93,39 +87,91 @@ jobs:
...
+          at: ~/nta
+      - run:
+          name: Validate Version
+          command: |
+            MY_VERSION=$(<dist/VERSION)
+            if [ "$MY_VERSION" != "$CIRCLE_TAG" ]; then
+              echo "Package version does not match release tag."
+              echo "Found release tag $CIRCLE_TAG but expected $MY_VERSION"
+              exit 1;
+            fi
```
Comment (rhyolight) [inline .circleci/config.yml]: I have not tested this.

**Pull request merged**